src/fireedge/src/modules/components/Tables/VrTemplates/DatastoreSelectionDialog.js

```diff
@@ -0,0 +1,170 @@
+ * a copy of the License at                                                  *
+ *                                                                           *
+ * http://www.apache.org/licenses/LICENSE-2.0                                *
+ *                                                                           *
+ * Unless required by applicable law or agreed to in writing, software       *
+ * distributed under the License is distributed on an "AS IS" BASIS,         *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  *
+ * See the License for the specific language governing permissions and       *
+ * limitations under the License.                                            *
+ * ------------------------------------------------------------------------- */
+import PropTypes from 'prop-types'
+import { useState, useMemo } from 'react'
+import {
+  Component,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  TextField,
+  Button,
+  DialogActions,
+  Box,
+  useTheme,
+} from '@mui/material'
+import { DatastoreCard } from '@modules/components/Cards'
+import { rowStyles } from '@modules/components/Tables/styles'
+
+/**
+ * @param {object} root0 - Props
+ * @param {boolean} root0.open - Is dialog open?
+ * @param {Function} root0.onClose - On close handler
+ * @param {Array} root0.datastores - Datastores
+ * @param {Function} root0.onSelect - On Select hanlder
+ * @param {boolean} root0.submitDisabled - Disable submit
+ * @returns {Component} - Datastore selection dialog
+ */
+export const DatastoreDialog = ({
+  open,
+  onClose,
+  datastores,
+  submitDisabled,
+  onSelect,
+}) => {
+  const theme = useTheme()
+  const classes = useMemo(() => rowStyles(theme), [theme])
+
+  const [searchTerm, setSearchTerm] = useState('')
+  const [currentPage, setCurrentPage] = useState(0)
+  const [selectedDatastore, setSelectedDatastore] = useState(null)
+  const itemsPerPage = 5
+
+  const filteredDatastores = datastores?.filter((datastore) =>
+    datastore.NAME.toLowerCase().includes(searchTerm.toLowerCase())
+  )
+
+  const paginatedDatastores = filteredDatastores.slice(
+    currentPage * itemsPerPage,
+    (currentPage + 1) * itemsPerPage
+  )
+
+  const handleSearchChange = (event) => {
+    setSearchTerm(event.target.value)
+    setCurrentPage(0)
+  }
+
+  const handleSelectDatastore = (datastore) => {
+    setSelectedDatastore(datastore)
+  }
+
+  const handleConfirmSelection = () => {
+    onSelect(selectedDatastore)
+    onClose()
+  }
+
+  const handlePrevPage = () => {
+    setCurrentPage((prevPage) => Math.max(prevPage - 1, 0))
+  }
+
+  const handleNextPage = () => {
+    setCurrentPage((prevPage) =>
+      Math.min(
+        prevPage + 1,
+        Math.ceil(filteredDatastores.length / itemsPerPage) - 1
+      )
+    )
+  }
+
+  return (
+    <Dialog open={open} onClose={onClose} maxWidth={'md'} fullWidth>
+      <DialogTitle>Select datastore for importing image</DialogTitle>
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

JSX element content is not internationalized. Application should support multiple languages through internationalization. Instead, utilize packages like &#39;i18next&#39; to internationalize your items.